### PR TITLE
feat(atomic): added logic to send analytics when opening user actions

### PR DESCRIPTION
**Pull request by mmitiche**
## [SFINT-5757](https://coveord.atlassian.net/browse/SFINT-5757)

In this PR:

- adding the logic log analytics when opening the user actions timeline feature in the insight panel.
- adding the property `excludedCustomActions` in the Atomic Insight User Actions components and passing to the Headless controller to allow the atomic user actions component to benefit from the already existing feature in the Headless user actions controller that allows filtering out unwanted events.

Demo showing the analytics being sent with `legacy` analytics:

https://github.com/user-attachments/assets/d66437d7-131f-4b42-bfbb-11ad0a6162ec

[SFINT-5757]: https://coveord.atlassian.net/browse/SFINT-5757?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by alexprudhomme — APPROVED** [review]
Seem like the tests I added today in my `did you mean` PR are flaky :/ . Sorry i'll fix it.

File: packages/atomic/src/components/insight/user-actions/atomic-insight-user-actions-modal/atomic-insight-user-actions-modal.tsx [hunk 1]
```diff
@@ -43,6 +43,10 @@ export class AtomicInsightUserActionsModal
    * The date and time when the case was created. For example "2024-01-01T00:00:00Z"
    */
   @Prop() public ticketCreationDateTime!: string;
+  /**
+   * The names of custom events to exclude.
+   */
+  @Prop() public excludedCustomActions: string[] = [];
```
Comment (fbeaudoincoveo) [inline packages/atomic/src/components/insight/user-actions/atomic-insight-user-actions-modal/atomic-insight-user-actions-modal.tsx:49]: I wonder: is "custom" the correct / future-proof term to use here?

The event protocol doesn't support custom events at all, so any event we emit will ultimately have its own schema. Hence there will be a definite set of actions that can be excluded, and this while this list may evolve over time, it most likely won't be extendable by individual customers.

Should we instead name the prop `excludedUserActions`, and document it like: "The names of the user action events to exclude"?

Also, is there already a list of possible values we could provide in the documentation?... I personally would have no clue what to pass to this prop.

Comment (mmitiche) [reply packages/atomic/src/components/insight/user-actions/atomic-insight-user-actions-modal/atomic-insight-user-actions-modal.tsx:49]: Very goof point indeed!

For now the scope of the user actions feature is really focused on the legacy analytics, our aim in Service is to really offer the same user actions capability for  our clients like they used to benefit from it using previous implementations built with the JSUI, that's why the current implementation is purposefully designed for the legacy analytics.

 The support for the user actions based on analytics logged with EP will come later, probably in ui-kit v4, as currently in the v3, the legacy analytics are still the standard in Service as EP is not fully ready to be used in any Service implementation.

**Review by lbergeron — APPROVED** [review]
I agree with @fbeaudoincoveo comment. Otherwise it looks good to me.

**Pull request merged**